skxray/fitting/xrf_model.py

```diff
@@ -0,0 +1,965 @@
+            # check if the line is activated
+            if linename not in self.element_line_name:
+                continue
+            new_width = element_dict['width'].copy()
+            if option:
+                new_width['adjust_element'] = option
+            self.new_parameter.update({str(linename)+'_delta_sigma': new_width})
+
+    def set_area(self, item, option=None):
+        """
+        Only the primary peak intensity is adjusted.
+
+        Parameters
+        ----------
+        item : str
+            element name
+        option : str, optional
+            way to control area
+        """
+        if item in k_line:
+            area_list = [str(item)+"_ka1_area"]
+        elif item in l_line:
+            item = item.split('_')[0]
+            area_list = [str(item)+"_la1_area"]
+
+        print('which item: {}'.format(item))
+
+        for linename in area_list:
+            new_area = element_dict['area'].copy()
+            if option:
+                new_area['adjust_element'] = option
+            self.new_parameter.update({linename: new_area})
+
+    def set_ratio(self, item, option=None):
+        """
+        Only adjust lines after the the primary one.
+
+        Parameters
+        ----------
+        item : str
+            element name
+        option : str, optional
+            way to control branching ratio
+        """
+        if item in k_line:
+            data_list = k_list[1:]
+        elif item in l_line:
+            item = item.split('_')[0]
+            data_list = l_list[1:]
+
+        data_list = [str(item)+'_'+str(v).lower() for v in data_list]
+
+        for linename in data_list:
+            if linename not in self.element_line_name:
+                continue
+            new_val = element_dict['ratio'].copy()
+            if option:
+                new_val['adjust_element'] = option
+            self.new_parameter.update({str(linename)+'_ratio_adjust': new_val})
+
+
+def get_sum_area(element_name, result_val):
+    """
+    Return the total area for given element.
+
+    Parameters
+    ----------
+    element_name : str
+        name of given element
+    result_val : obj
+        result obj from lmfit to save all the fitting results
+
+    Returns
+    -------
+    float
+        the total area
+    """
+    def get_value(result_val, element_name, line_name):
+        return result_val.values[str(element_name)+'_'+line_name+'_area'] * \
+               result_val.values[str(element_name)+'_'+line_name+'_ratio'] * \
+               result_val.values[str(element_name)+'_'+line_name+'_ratio_adjust']
+
+    if element_name in k_line:
+        sum = get_value(result_val, element_name, 'ka1') + \
+              get_value(result_val, element_name, 'ka2') + \
+              get_value(result_val, element_name, 'kb1')
+        if result_val.values.has_key(str(element_name)+'_kb2_area'):
+            sum += get_value(result_val, element_name, 'kb2')
+    return sum
```

Comment:
There is a code path where this is not defined and will raise a `NotDefined` exception which is not super useful to the user.